src/libprojectM/Renderer/MilkdropNoise.cpp

```diff
@@ -0,0 +1,262 @@
+#include "MilkdropNoise.hpp"
+
+#include <cmath>
+#include <ctime>
+
+MilkdropNoise::MilkdropNoise()
+{
+    generate2D(256, 1, &noise_lq);
+    generate2D(32, 1, &noise_lq_lite);
+    generate2D(256, 4, &noise_mq);
+    generate2D(256, 8, &noise_hq);
+
+    generate3D(32, 1, &noise_lq_vol);
+    generate3D(32, 4, &noise_hq_vol);
+}
+
+MilkdropNoise::~MilkdropNoise()
+{
+    delete[] noise_lq;
+    delete[] noise_lq_lite;
+    delete[] noise_mq;
+    delete[] noise_hq;
+    delete[] noise_lq_vol;
+    delete[] noise_hq_vol;
+}
+
+void MilkdropNoise::generate2D(int size, int zoomFactor, uint32_t** textureData)
+{
+    *textureData = new uint32_t[size * size]();
+
+    // write to the bits...
+    auto dst = *textureData;
+    auto RANGE = (zoomFactor > 1) ? 216 : 256;
+    srand(time(nullptr));
```

Comment:
don't think we need to srand every time do we?

Reply:
Probably not. Milkdrop does it though, just with a different RNG from the Windows API.

Maybe I should better convert this to std::random instead.

Reply:
Done. Decided to still create a PRNG in each function to skip the need to either declare two additional class members or a parameter to pass on the generator into the noise functions. Distribution and random seed should be better now (not that it is highly relevant for a noise texture).